core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -549,18 +556,23 @@ public void handleClientOperation(int operation) throws Exception {
 
     @Override
     public PGConnectionContext of(int fd, @NotNull IODispatcher<PGConnectionContext> dispatcher) {
-        super.of(fd, dispatcher);
-        sqlExecutionContext.with(fd);
-        if (recvBuffer == 0) {
-            this.recvBuffer = Unsafe.malloc(recvBufferSize, MemoryTag.NATIVE_PGW_CONN);
-        }
-        if (sendBuffer == 0) {
-            this.sendBuffer = Unsafe.malloc(sendBufferSize, MemoryTag.NATIVE_PGW_CONN);
-            this.sendBufferPtr = sendBuffer;
-            this.sendBufferLimit = sendBuffer + sendBufferSize;
-        }
-        authenticator.init(socket, recvBuffer, recvBuffer + recvBufferSize, sendBuffer, sendBufferLimit);
-        return this;
+        try {
+            super.of(fd, dispatcher);
+            sqlExecutionContext.with(fd);
```

Comment:
same story, `of` attempts to reuse the context, if that is not successful the object should be deposited back to pool without being closed